pkg/api/v1/routes/handlers.go

```diff
@@ -0,0 +1,417 @@
+
+		return
+	}
+
+	// query existing condition
+	existing, err := r.repository.Get(c.Request.Context(), serverID, kind)
+	if err != nil {
+		c.JSON(
+			http.StatusBadRequest,
+			&v1types.ServerResponse{Message: err.Error()},
+		)
+
+		return
+	}
+
+	if existing == nil {
+		c.JSON(
+			http.StatusNotFound,
+			&v1types.ServerResponse{Message: "no existing condition found for update"},
+		)
+
+		return
+	}
+
+	// nothing to update
+	if existing.State == conditionUpdate.State && bytes.Equal(existing.Status, conditionUpdate.Status) {
+		c.JSON(http.StatusOK, &v1types.ServerResponse{Message: "no changes to be applied"})
+
+		return
+	}
+
+	// merge update with existing
+	update, err := conditionUpdate.MergeExisting(existing)
+	if err != nil {
+		c.JSON(
+			http.StatusBadRequest,
+			&v1types.ServerResponse{Message: err.Error()},
+		)
+
+		return
+	}
+
+	// update
+	if err := r.repository.Update(c.Request.Context(), serverID, update); err != nil {
+		c.JSON(
+			http.StatusInternalServerError,
+			&v1types.ServerResponse{Message: err.Error()},
+		)
+
+		return
+	}
+
+	c.JSON(http.StatusOK, &v1types.ServerResponse{Message: "condition updated"})
+}
+
+func (r *Routes) serverConditionCreate(c *gin.Context) {
+	serverID, err := uuid.Parse(c.Param("uuid"))
+	if err != nil {
+		c.JSON(
+			http.StatusBadRequest,
+			&v1types.ServerResponse{Message: err.Error()},
+		)
+
+		return
+	}
+
+	kind := ptypes.ConditionKind(c.Param("conditionKind"))
+	if !r.conditionKindValid(kind) {
+		c.JSON(
+			http.StatusBadRequest,
+			&v1types.ServerResponse{Message: "unsupported condition kind: " + string(kind)},
+		)
+
+		return
+	}
+
+	var conditionCreate v1types.ConditionCreate
+	if err = c.ShouldBindJSON(&conditionCreate); err != nil {
+		c.JSON(
+			http.StatusBadRequest,
+			&v1types.ServerResponse{Message: "invalid ConditionCreate payload: " + err.Error()},
+		)
+
+		return
+	}
+
+	condition := conditionCreate.NewCondition(kind)
+
+	// check the condition doesn't already exist in a non-finalized state
+	existing, err := r.repository.Get(c.Request.Context(), serverID, kind)
+	if err != nil && !errors.Is(err, store.ErrConditionNotFound) {
+		c.JSON(
+			http.StatusBadRequest,
+			&v1types.ServerResponse{Message: err.Error()},
+		)
+
+		return
+	}
+
+	if existing != nil && !ptypes.ConditionStateFinalized(existing.State) {
+		c.JSON(
+			http.StatusBadRequest,
+			&v1types.ServerResponse{Message: "condition present non-finalized state: " + string(existing.State)},
+		)
+
+		return
+	}
+
+	// purge the existing condition
+	if existing != nil {
+		err = r.repository.Delete(c.Request.Context(), serverID, kind)
+		if err != nil && !errors.Is(err, store.ErrConditionNotFound) {
+			c.JSON(
+				http.StatusInternalServerError,
+				&v1types.ServerResponse{Message: err.Error()},
+			)
+
+			return
+		}
+	}
+
+	// check if any condition with exclusive set is in non-finalized states
+	if errEx := r.exclusiveNonFinalConditionExists(c.Request.Context(), serverID); errEx != nil {
```

Comment:
This was split into its own function since the linter was complaining about the cyclomatic complexity being high for [serverConditionCreate()](https://github.com/metal-toolbox/conditionorc/blob/mvp/pkg/api/v1/routes/handlers.go#L130), unless you mean't something else here - I'm inclined to leave it as is for now..

Reply:
I'm OK with leaving this for now. I'll try and refactor a bit, maybe, and see if I can't appease the linter.